### ISSUE TITLE
AttributeError: '_Code' object has no attribute 'co_consts'

### ISSUE THREAD

**Issue by ba1dr** [Delgan/loguru#52]
loguru==0.2.5

    [2019-02-06 17:44:42,144: WARNING/ForkPoolWorker-2] --- Logging error in Loguru Handler #0 ---
    [2019-02-06 17:44:42,145: WARNING/ForkPoolWorker-2] Record was:
    [2019-02-06 17:44:42,145: WARNING/ForkPoolWorker-2] {'elapsed': datetime.timedelta(0, 3, 912801), 'exception': <loguru._recattrs.ExceptionRecattr object at 0x7fde0b437ba8>, 'extra': {}, 'file': 'logging_settings.py', 'function': 'emit', 'level': 'Level 40', 'line': 15, 'message': 'Task epex.tasks.run_user_worker[012ace7d-8264-4ac9-8305-0c243e20e8bc] raised unexpected: NameError("name \'blabla\' is not defined",)', 'module': 'logging_settings', 'name': 'navydog.settings.logging_settings', 'process': '20371', 'thread': '140591889479488', 'time': datetime(2019, 2, 6, 17, 44, 42, 144557, tzinfo=datetime.timezone(datetime.timedelta(0), 'UTC'))}
    [2019-02-06 17:44:42,146: WARNING/ForkPoolWorker-2] Traceback (most recent call last):
    [2019-02-06 17:44:42,147: WARNING/ForkPoolWorker-2] File "/path/to/project/venv/lib/python3.6/site-packages/celery/app/trace.py", line 382, in trace_task
        R = retval = fun(*args, **kwargs)
    [2019-02-06 17:44:42,147: WARNING/ForkPoolWorker-2] File "/path/to/project/venv/lib/python3.6/site-packages/celery/app/trace.py", line 641, in __protected_call__
        return self.run(*args, **kwargs)
    [2019-02-06 17:44:42,147: WARNING/ForkPoolWorker-2] File "/path/to/project/navydog/apps/epex/tasks.py", line 37, in run_user_worker
        blabla(epex_user)
    [2019-02-06 17:44:42,147: WARNING/ForkPoolWorker-2] NameError: name 'blabla' is not defined
    [2019-02-06 17:44:42,147: WARNING/ForkPoolWorker-2] During handling of the above exception, another exception occurred:
    [2019-02-06 17:44:42,147: WARNING/ForkPoolWorker-2] Traceback (most recent call last):
    [2019-02-06 17:44:42,148: WARNING/ForkPoolWorker-2] File "/path/to/project/venv/lib/python3.6/site-packages/loguru/_handler.py", line 174, in emit
        error = exception.format_exception(self.backtrace, self.colorize, self.encoding)
    [2019-02-06 17:44:42,148: WARNING/ForkPoolWorker-2] File "/path/to/project/venv/lib/python3.6/site-packages/loguru/_recattrs.py", line 160, in format_exception
        self._extended_traceback = self._extend_traceback(traceback_, self._decorated)
    [2019-02-06 17:44:42,148: WARNING/ForkPoolWorker-2] File "/path/to/project/venv/lib/python3.6/site-packages/loguru/_recattrs.py", line 92, in _extend_traceback
        tb = self._make_catch_traceback(tb.tb_frame, tb.tb_lasti, tb.tb_lineno, tb.tb_next)
    [2019-02-06 17:44:42,148: WARNING/ForkPoolWorker-2] File "/path/to/project/venv/lib/python3.6/site-packages/loguru/_recattrs.py", line 119, in _make_catch_traceback
        c.co_consts,
    [2019-02-06 17:44:42,148: WARNING/ForkPoolWorker-2] AttributeError: '_Code' object has no attribute 'co_consts'
    [2019-02-06 17:44:42,148: WARNING/ForkPoolWorker-2] --- End of logging error ---

**Comment by Delgan** [comment]
Hey @ba1dr, thanks for the bug report.

Are you able to give me a code snippet so I could reproduce and investigate the error, please?

**Comment by ba1dr** [comment]
Hi,
taking it out of the context the code might be like below. I use `Django`/`celery` project and this is how loggers initialized. I have already rewritten that part and made it working another way, so restoring it by memory.

This is how it occurs in my code, but I am pretty sure it might be reproduced even without celery and signals.

```

import logging
from celery import shared_task
from celery.signals import after_setup_logger
from loguru import logger as loguru_logger

class LoguruInterceptHandler(logging.Handler):
    def emit(self, record):
        logger_opt = loguru_logger.opt(depth=100, exception=record.exc_info)
        logger_opt.log(record.levelno, record.getMessage())

@after_setup_logger.connect
def on_celery_setup_logging(logger=None, loglevel=logging.DEBUG, **kwargs):
    handler = LoguruInterceptHandler()
    handler.setLevel(logging.ERROR)
    logger.addHandler(handler)

@shared_task
def mytask():
    variable = 123
    blabla(variable)  # NameError occurs here

```

Then you run celery worker and send the task to worker.

I am sorry I did not try to reproduce it in more clean environment (without celery). I thought you maybe could find uninitialized attribute by code yourself.

**Comment by Delgan** [comment]
Thank you for taking the time to write this!

Don't worry, I will find a way to reproduce it.
But it's weird, it usually works well. This `_Code` object in your stacktrace is doubtful, I don't know from where it comes from.

**Comment by Delgan** [comment]
So, I finally merged the fix for this, this should be available in the next `v0.3.0` release.

The error came from `billiard` which is a dependency of `celery`. In some circunstances, the exception stack trace is populated with a custom `_Code` object which lacks the `co_consts` attribute used by Loguru: [`billiard/einfo.py`](https://github.com/celery/billiard/blob/529a3d59c50528c21d68c2e2bc176cefae410716/billiard/einfo.py#L11-L26).
`billiard` is not the only library to implement their own traceback objects, see Twisted: [`twisted/failure.py`](https://github.com/twisted/twisted/blob/29cbea324e1b1bb1d38863fbc0ba640d4d218177/src/twisted/python/failure.py#L175-L181).

I fixed it by completely refactoring the way Loguru formats exceptions, there was no need to create our own `code` objects too, so there is no need to access `co_consts` and others such attributes.
